great_expectations/datasource/fluent/spark_filesystem_datasource.py

```diff
@@ -56,20 +61,33 @@ def test_connection(self, test_assets: bool = True) -> None:
                 asset.test_connection()
 
     def _build_data_connector(
-        self, data_asset: CSVAsset, glob_directive: str = "**/*", **kwargs
+        self,
+        data_asset: _SPARK_FILE_PATH_ASSET_TYPES,
+        glob_directive: str = "**/*",
+        **kwargs,
     ) -> None:
         """Builds and attaches the `FilesystemDataConnector` to the asset."""
         if kwargs:
             raise TypeError(
                 f"_build_data_connector() got unexpected keyword arguments {list(kwargs.keys())}"
             )
+        if hasattr(data_asset, "data_directory"):
```

Comment:
Could we do an isinstance check here? It makes it clearer that we are matching over the possible types of the `data_asset`.

Reply:
Agreed. I was trying to avoid keeping a tuple of directory related asset classes, but I agree this is clearer.